build-system/tasks/dist.js

```diff
@@ -103,42 +105,54 @@ function printDistHelp() {
  * Perform the prerequisite steps before starting the minified build.
  * Used by `gulp` and `gulp dist`.
  *
- * @param {boolean} watch
+ * @param {!Object} options
  */
-async function runPreDistSteps(watch) {
+async function runPreDistSteps(options) {
   cleanupBuildDir();
   await prebuild();
-  await compileCss(watch);
+  await compileCss(options);
   await compileJison();
   await copyCss();
   await copyParsers();
-  await bootstrapThirdPartyFrames(watch, /* minify */ true);
+  await bootstrapThirdPartyFrames(options);
   await startNailgunServer(distNailgunPort, /* detached */ false);
   displayLifecycleDebugging();
 }
 
 /**
- * Dist Build
- * @return {!Promise}
+ * Minified build. Entry point for `gulp dist`.
  */
 async function dist() {
```

Comment:
nit/question: why a separate `doDist()`?

Reply:
This is because `dist()` is the entry point for `gulp dist` and receives an unused `done` callback when you directly invoke the task. Therefore, we can't add any params to `dist()`. Hence the separate `doDist()`.